code/game/objects/items/devices/radio/intercom.dm

```diff
@@ -139,6 +139,12 @@
 /obj/item/radio/intercom/add_blood_DNA(list/blood_dna)
 	return FALSE
 
+/obj/item/radio/intercom/end_emp_effect(curremp)
+	. = ..()
+	if(!.)
+		return
+	on = TRUE

```

Comment:
I don't believe you converted this behavior correctly, it used to set `on = TRUE` for everything except intercoms. Here it sets `on = TRUE` for intercoms and nothing else.